# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../ocfs2-tools/ocfs2-tools-1.4-glibc.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Patch origin: http://oss.oracle.com/pipermail/ocfs2-tools-devel/2010-March/002863.html

diff -ruN ocfs2-tools-1.4.4-orig/fsck.ocfs2/Makefile ocfs2-tools-1.4.4/fsck.ocfs2/Makefile
--- ocfs2-tools-1.4.4-orig/fsck.ocfs2/Makefile	2010-03-17 01:10:43.000000000 +0100
+++ ocfs2-tools-1.4.4/fsck.ocfs2/Makefile	2010-07-28 11:03:51.772258975 +0200
@@ -17,6 +17,7 @@
 
 ifndef OCFS2_DYNAMIC_FSCK
 LDFLAGS += -static
+STATICLIBS=-lpthread
 endif
 
 CFILES =	fsck.c		\
@@ -66,7 +67,7 @@
 	$(TOPDIR)/mkinstalldirs $(DIST_DIR)/include
 
 fsck.ocfs2: $(OBJS) $(LIBOCFS2_DEPS) $(LIBO2DLM_DEPS) $(LIBO2CB_DEPS)
-	$(LINK) $(LIBOCFS2_LIBS) $(LIBO2DLM_LIBS) $(LIBO2CB_LIBS) $(COM_ERR_LIBS)
+	$(LINK) $(LIBOCFS2_LIBS) $(LIBO2DLM_LIBS) $(LIBO2CB_LIBS) $(COM_ERR_LIBS) $(STATICLIBS)
 
 $(OBJS): prompt-codes.h
 
diff -ruN ocfs2-tools-1.4.4-orig/o2cb_ctl/Makefile ocfs2-tools-1.4.4/o2cb_ctl/Makefile
--- ocfs2-tools-1.4.4-orig/o2cb_ctl/Makefile	2008-12-03 19:38:22.000000000 +0100
+++ ocfs2-tools-1.4.4/o2cb_ctl/Makefile	2010-07-28 11:03:51.772258975 +0200
@@ -15,6 +15,7 @@
 
 ifndef OCFS2_DYNAMIC_CTL
 LDFLAGS += -static
+STATICLIBS=-lpthread
 endif
 
 DEFINES = -DVERSION=\"$(VERSION)\"
@@ -40,6 +41,6 @@
 o2cb_ctl_CPPFLAGS = $(GLIB_CFLAGS) -DG_DISABLE_DEPRECATED
 
 o2cb_ctl: $(O2CB_CTL_OBJS) $(LIBOCFS2_DEPS) $(LIBO2CB_DEPS)
-	$(LINK) $(LIBO2CB_LIBS) $(GLIB_LIBS) $(LIBOCFS2_LIBS) $(COM_ERR_LIBS)
+	$(LINK) $(LIBO2CB_LIBS) $(GLIB_LIBS) $(LIBOCFS2_LIBS) $(COM_ERR_LIBS) $(STATICLIBS)
 
 include $(TOPDIR)/Postamble.make
diff -ruN ocfs2-tools-1.4.4-orig/ocfs2_hb_ctl/Makefile ocfs2-tools-1.4.4/ocfs2_hb_ctl/Makefile
--- ocfs2-tools-1.4.4-orig/ocfs2_hb_ctl/Makefile	2008-12-03 19:38:22.000000000 +0100
+++ ocfs2-tools-1.4.4/ocfs2_hb_ctl/Makefile	2010-07-28 11:03:51.772258975 +0200
@@ -15,6 +15,7 @@
 
 ifndef OCFS2_DYNAMIC_CTL
 LDFLAGS += -static
+STATICLIBS=-lpthread
 endif
 
 DEFINES = -DVERSION=\"$(VERSION)\"
@@ -32,6 +33,6 @@
 all: ocfs2_hb_ctl
 
 ocfs2_hb_ctl: $(OBJS) $(LIBOCFS2_DEPS) $(LIBO2DLM_DEPS) $(LIBO2CB_DEPS)
-	$(LINK) $(LIBOCFS2_LIBS) $(LIBO2DLM_LIBS) $(LIBO2CB_LIBS) $(COM_ERR_LIBS)
+	$(LINK) $(LIBOCFS2_LIBS) $(LIBO2DLM_LIBS) $(LIBO2CB_LIBS) $(COM_ERR_LIBS) $(STATICLIBS)
 
 include $(TOPDIR)/Postamble.make
